TruePath/PathExtensions.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TruePath;
+/// <summary>
+/// Extension methods for <see cref="IPath"/> and <see cref="IPath{TPath}"/>.
+/// </summary>
+public static class PathExtensions
+{
+    /// <summary>
+    /// Gets the extension of the file name of the path.
+    /// </summary>
+    /// <typeparam name="TPath"></typeparam>
+    /// <param name="path"></param>
+    /// <returns></returns>
+    public static string GetExtension(this IPath path)
+    {
+        var fileExtenstion = Path.GetExtension(path.FileName);
+        if (string.IsNullOrEmpty(fileExtenstion))
+        {
+            throw new ArgumentException(string.Format("{0} is not a file", path.FileName));
+        }
+        return fileExtenstion;
+    }
```

Comment:
I propose the following amendments.
1. Let's make this return `null` on paths without extensions.
2. Let's make this return a string consisting of only a dot on paths with the empty extension (i.e. whose names _end with dot_, even though it's unusual) — and add test.
3. Let's not throw an exception in any situation.
4. Let's rename this into `GetExtensionWithDot()`.
   **Why?** I see no consensus among other libraries on whether dot should or should not be included by default. In .NET, it is with dot; in Kotlin, it is without. Different methods are useful in different cases, so let's just have two, both explicitly explaining whether they will or will not include the dot.